infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/fsm/LogReplicationFSM.java

```diff
@@ -253,6 +253,9 @@ public LogReplicationFSM(CorfuRuntime runtime, SnapshotReader snapshotReader, Da
         // Initialize Log Replication 5 FSM states - single instance per state
         initializeStates(snapshotSender, logEntrySender, dataSender);
 
+        // Assign logEntryReader to ackReader
+        this.ackReader.setLogEntryReader(logEntryReader);
```

Comment:
I think we might have an issue here, please look at the comment in LogReplicationAckReader with regards to this.